### PR TITLE
Don't fail seeing a backup refering to non-existent disk

### PR DESCRIPTION
**Pull request by ydirson**
This is the code submitted as xenserver/host-installer#70 (formerly #67)

### CODE REVIEW

**Review by benjamreis — APPROVED** [review]

File: diskutil.py [hunk 1]
```diff
@@ -280,6 +280,9 @@ def getDiskDeviceSize(dev):
         return int(__readOneLineFile__("/sys/block/%s/device/block/size" % dev))
     elif os.path.exists("/sys/block/%s/size" % dev):
         return int(__readOneLineFile__("/sys/block/%s/size" % dev))
+    else:
+        assert 0, "%s not found as %s or %s" % (dev, "/sys/block/%s/device/block/size",
```
Comment (stormi) [inline diskutil.py]: Why `assert 0` rather than raising an exception directly?

File: product.py [hunk 2]
```diff
@@ -537,9 +537,17 @@ def findXenSourceBackups():
...
+                    if backup.version < XENSERVER_MIN_VERSION:
+                        logger.log("findXenSourceBackups: ignoring, platform too old: %s < %s" %
+                                   (backup.version, XENSERVER_MIN_VERSION))
+                        break
+                    if backup.version > THIS_PLATFORM_VERSION:
+                        logger.log("findXenSourceBackups: ignoring later platform: %s > %s" %
+                                   (backup.version, THIS_PLATFORM_VERSION))
+                        break
                     backups.append(backup)
+                    break
```
Comment (stormi) [inline product.py]: I might be missing something, but at first sight this looks a lot like a complicated way to do what `if... elif... else`  would do. I see no way to actually loop a second time in this `while` loop.

Comment (ydirson) [reply product.py]: The loop is just here as a `try: finally:` without any exception overhead. But right, it's overkill here.

File: diskutil.py (hunk 1 above)
Comment (ydirson) [reply diskutil.py]: A cursory glance was showing several `assert 0` already, but they are a different kind indeed, and even a buggy use of assert.
Good point.

File: product.py [hunk 3]
```diff
@@ -537,8 +537,13 @@ def findXenSourceBackups():
             if os.path.exists(os.path.join(b.mount_point, '.xen-backup-partition')):
                 backup = XenServerBackup(p, b.mount_point)
                 logger.log("Found a backup: %s" % (repr(backup),))
-                if backup.version >= XENSERVER_MIN_VERSION and \
-                        backup.version <= THIS_PLATFORM_VERSION:
+                if backup.version < XENSERVER_MIN_VERSION:
+                    logger.log("findXenSourceBackups: ignoring, platform too old: %s < %s" %
+                                   (backup.version, XENSERVER_MIN_VERSION))
```
Comment (stormi) [inline product.py]: Should be indented like line 545?

Comment (ydirson) [reply product.py]: At first I thought your remark was about the different indent wrt `if` of line 537.
This indeed made me focus on the role of these nested if's: as an extra condition for the whole nominal path, it makes a case for getting back to the lightweight `try:except` in my original refactor (idiom akin to `goto fail` error paths in C)

**Review by stormi — APPROVED** [review]

**Review by stormi — APPROVED** [review]

**Pull request merged**